test/optim/test_inplace_activations.py

```diff
@@ -0,0 +1,110 @@
+"""SINGD with a model that uses in-place activations."""
+
+from copy import deepcopy
+from test.utils import REDUCTION_IDS, REDUCTIONS, compare_optimizers
+
+from pytest import mark, skip
+from torch import manual_seed, rand
+from torch.nn import Conv2d, CrossEntropyLoss, Flatten, Linear, ReLU, Sequential
+from torch.utils.data import DataLoader
+from torchvision.datasets import MNIST
+from torchvision.transforms import ToTensor
+
+from singd.optim.optimizer import SINGD
+
+
+@mark.parametrize("inplace", [True, False], ids=["inplace=True", "inplce=False"])
+def test_hooks_support_inplace_activations(inplace: bool):
+    """Test that SINGD's hooks support in in-place activations.
+
+    See https://github.com/f-dangel/singd/issues/56.
+
+    Args:
+        inplace: Whether to use in-place activations.
+    """
+    manual_seed(0)
+
+    X = rand(2, 1, 5, 5)
+    model = Sequential(Conv2d(1, 1, 3), ReLU(inplace=inplace))
+    SINGD(model)  # install hooks
+
+    model(X)
+
+
+@mark.parametrize("reduction", REDUCTIONS, ids=REDUCTION_IDS)
+def test_compare_training_using_inplace_activations(reduction: str):
+    """Compare training w/o in-place activations.
+
+    Args:
+        reduction: Reduction used for the loss function.
+    """
+    if reduction == "sum":
+        skip("Need to fix https://github.com/f-dangel/singd/issues/43 first.")
+
+    manual_seed(0)
+    MAX_STEPS = 100
+    batch_size = 32
+
+    train_dataset = MNIST("./data", train=True, download=True, transform=ToTensor())
+    train_loader = DataLoader(
+        dataset=train_dataset, batch_size=batch_size, shuffle=True
+    )
+
+    # _inplace indicates that the trained net has in-place activations
+
+    # NOTE All parameters of this net are supported by SINGD, no optimizer is involved
```

Comment:
"(...), no _other_ optimizer is involved"?

Reply:
Removed that.